bundles/org.eclipse.equinox.http.servlet/src/org/eclipse/equinox/http/servlet/internal/context/DispatchTargets.java

```diff
@@ -119,6 +119,20 @@ else if (dispatcherType == DispatcherType.FORWARD) {
 
 			requestWrapper.push(this);
 
+			if (dispatcherType == DispatcherType.INCLUDE) {
+				HttpServletResponse previous = null;
+				HttpServletResponse next = response;
+
+				while (next instanceof HttpServletResponseWrapper) {
+					previous = next;
+					next = (HttpServletResponse) ((HttpServletResponseWrapper) next).getResponse();
+				}
```

Comment:
Don't we already have the initial response as `responseWrapper`? Why don't we just do:

```java
responseWrapper.setResponse(new IncludeDispatchResponseWrapper(responseWrapper.getResonse()));
```
_omitted casts for clarity_